homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,124 @@
+
+import logging
+from typing import Any
+
+from homeassistant import config_entries, core
+from homeassistant.components.sensor import STATE_CLASS_MEASUREMENT, SensorEntity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import ATTR_COUNTY, ATTRIBUTION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS = {
+    "count": "mdi:virus",
+    "deaths": "mdi:cross",
+    "recovered": "mdi:bottle-tonic-plus-outline",
+    "week_incidence": "mdi:clipboard-pulse",
+    "cases_per100k": "mdi:home-group",
+    "new_cases": "mdi:shield-bug",
+    "new_deaths": "mdi:shield-cross",
+    "new_recovered": "mdi:shield-sync",
+}
+
+DISTRICT_SENSORS = {
+    "hospitalization_cases_baby": "mdi:baby-face-outline",
+    "hospitalization_incidence_baby": "mdi:baby-face",
+    "hospitalization_cases_children": "mdi:account-child-outline",
+    "hospitalization_incidence_children": "mdi:account-child",
+    "hospitalization_cases_teen": "mdi:face-woman",
+    "hospitalization_incidence_teen": "mdi:face-woman-outline",
+    "hospitalization_cases_grown": "mdi:face-man",
+    "hospitalization_incidence_grown": "mdi:face-man-outline",
+    "hospitalization_cases_senior": "mdi:account-cowboy-hat-outline",
+    "hospitalization_incidence_senior": "mdi:account-cowboy-hat",
+    "hospitalization_cases_old": "mdi:human-white-cane",
+    "hospitalization_incidence_old": "mdi:human-cane",
+}
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create sensors from a config entry in the integrations UI."""
+    _LOGGER.debug("create sensor from config entry %s", config_entry.data)
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    district = config_entry.data[ATTR_COUNTY]
+    sensors = [
+        RKICovidNumbersSensor(coordinator, district, info_type) for info_type in SENSORS
+    ]
+    async_add_entities(sensors, update_before_add=True)
+
+    # add additional sensors for districts
+    if district.startswith("BL"):
+        district_sensors = [
+            RKICovidNumbersSensor(coordinator, district, info_type)
+            for info_type in DISTRICT_SENSORS
+        ]
+        async_add_entities(district_sensors, update_before_add=True)
+
+
+class RKICovidNumbersSensor(SensorEntity, CoordinatorEntity):
+    """Representation of a sensor."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_state_class = STATE_CLASS_MEASUREMENT
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[Any],
+        district: dict[str, str],
+        info_type: str,
+    ) -> None:
+        """Initialize a new sensor."""
+        _LOGGER.debug("initialize %s sensor for %s", info_type, district)
+        super().__init__(coordinator)
+
+        data = coordinator.data[district]
```

Comment:
Seems a bit unnecessary as it's not used elsewhere than init.

Reply:
unnecessary to put it in a local variable? isn't it much more readable?